### ISSUE TITLE
On the issue of the non convergence  of model when using newer versions (==0.3.1) ？

### ISSUE THREAD

**Issue by cpt-via** [qubvel/segmentation_models.pytorch#742]
Because I want to use **in_ Channels** interface, so I upgraded my library version, I used the same training code and the same hyperparameters, when the library version **==0.0.3**, the model can converge quickly, as shown in log(1). However, when the library version is **==0.3.1**, the model does not converge, as shown in log(2). What is the possible reason?

Due to limited space, only the first 2 epochs are displayed.

**log(1):**
Starting epoch: 0 | phase: train | start time: 16:46:28
Loss: 0.86396092 | IoU: 0.148 | dice: 0.196 | tp: 340.1  | fp: 14061.2 | fn: 63.8
Starting epoch: 0 | phase: val | start time: 16:46:58
Loss: 0.76130650 | IoU: 0.273 | dice: 0.369 | tp: 342.0  | fp: 13207.3 | fn: 112.3

Starting epoch: 1 | phase: train | start time: 16:47:04
Loss: 0.73710251 | IoU: 0.282 | dice: 0.366 | tp: 323.9  | fp: 10247.5 | fn: 76.9
Starting epoch: 1 | phase: val | start time: 16:47:25
Loss: 0.69458284 | IoU: 0.337 | dice: 0.425 | tp: 344.3  | fp: 12329.2 | fn: 110.0

**log(2):**
Starting epoch: 0 | phase: train | start time: 17:23:38
Loss: 0.98351806 | IoU: 0.007 | dice: 0.012 | tp: 408.2  | fp: 65127.8 | fn: 0.0
Starting epoch: 0 | phase: val | start time: 17:23:58
Loss: 0.98160886 | IoU: 0.008 | dice: 0.014 | tp: 454.3  | fp: 65081.7 | fn: 0.0
******** New optimal found, saving state ********

Starting epoch: 1 | phase: train | start time: 17:24:02
Loss: 0.98348601 | IoU: 0.007 | dice: 0.012 | tp: 409.3  | fp: 65126.7 | fn: 0.0
Starting epoch: 1 | phase: val | start time: 17:24:16
Loss: 0.98160885 | IoU: 0.007 | dice: 0.014 | tp: 454.3  | fp: 65081.7 | fn: 0.0
******** New optimal found, saving state ********

**PS:**
Some library versions used by log (1):
torch==1.1.0
torchvision==0.4.0
efficientnet-pytorch==0.4.0
pretrainedmodels==0.7.4
segmentation-models-pytorch==0.0.3 

Some library versions used by log (2):
torch==1.7.0
torchvision==0.8.0
efficientnet-pytorch==0.7.1
pretrainedmodels==0.7.4
segmentation-models-pytorch==0.3.1

Or, if I want to use **in_Channels** interface, how can I change my code or change the library version? Thank you for your answer!

**Comment by qubvel** [comment]
Hi, probably some loss computation logic is changed, it is hard to guess. I can only suggest check your input parameters and code once again and use debugger to find an error
